File: bongf/python/level1/phoneketmon.py
def solution(nums):
    max = int( len(nums) / 2 )
    kind = len(set(nums))
    if kind >= max :
        return max
    else : 
        return kind

def solution_2(nums):
    return min(len(nums)/2, len(set(nums)))

nums = [3,3,3,2,2,2,2,2,2]
print(solution(nums))

File: programmers/java/src/level1/Phoneketmon.java
package level1;

import java.util.Arrays;
import java.util.stream.Collectors;

public class Phoneketmon {
    public static void main(String[] args) {
        int[] nums = {3,3,3,2,2,2};
        System.out.println(solution(nums));
    }

    public static int solution(int[] nums) {
        return Math.min( nums.length /2 ,  Arrays.stream(nums).boxed().collect(Collectors.toSet()).size());
    }

    public static int solution_2(int[] nums) {
        return Arrays.stream(nums)
                .boxed()
                .collect(Collectors.collectingAndThen(Collectors.toSet(),
                        x -> Integer.min(x.size(), nums.length / 2)));
    }
}
